pypsa/optimization/optimize.py

```diff
@@ -139,6 +139,31 @@ def define_objective(n, sns):
             objective.append((operation * operation * cost).sum())
             is_quadratic = True
 
+    # stand-by cost
+    comps = {"Generator", "Link"}
+    for c in comps:
+        com_i = get_committable_i(n, c)
+
+        if com_i.empty:
+            continue
+
+        stand_by_cost = (
+            get_as_dense(n, c, "stand_by_cost", sns)
+            .loc[:, lambda ds: (ds != 0).any()]
+            .mul(weighting, axis=0)
+        )
+
+        idx = stand_by_cost.columns
+        com_and_stand_by_cost = com_i.intersection(idx)
+        status = n.model.variables[f"{c}-status"].loc[:, com_and_stand_by_cost]
+        stand_by_cost.columns.set_names(
+            stand_by_cost.columns.name + "-com", inplace=True
+        )
+
+        m.objective = (
+            m.objective + (status * stand_by_cost.loc[:, com_and_stand_by_cost]).sum()
+        )
```

Comment:
```suggestion
        stand_by_cost = (
            get_as_dense(n, c, "stand_by_cost", sns, com_i)
            .loc[:, lambda ds: (ds != 0).any()]
            .mul(weighting, axis=0)
        )

        stand_by_cost.columns.name = f"{c}-com"
        status = n.model.variables[f"{c}-status"].loc[:, stand_by_cost.columns]
        m.objective = (
            m.objective + (status * stand_by_cost).sum()
        )
```
Could you try out whether this refined option works?

Reply:
and thanks btw, looks nearly perfect :)

Reply:
@enricogiglio could you try to run it?

Reply:
It seems to be working and thank you very much @FabianHofmann :)